Python/getversion.c

```diff
@@ -19,8 +19,17 @@ void _Py_InitVersion(void)
 #else
     const char *buildinfo_format = "%.80s (%.80s) %.80s";
 #endif
+    // The format string is defined above and is observably safe.
```

Comment:
Could it be switched to a `#define`d literal? That way it could look safe to the compiler, too.

Reply:
I just pushed a change. I figure instead of going through the trouble defining a format string in the preprocessor #if/#else blocks and adding all of the diagnostic pragmas we can just put the `PyOS_snprintf()` with the relevant format string literals in the respective #if/#else blocks. We eliminate the root cause of the warning instead of ignore it.